scheduler/scheduler.go

```diff
@@ -589,11 +590,19 @@ func (s *scheduler) gatherMetricsAndPlugins(wf *schedulerWorkflow) ([]core.Metri
 	)
 
 	for _, m := range wf.metrics {
-		mts = append(mts, &metric{
-			namespace: m.Namespace(),
-			version:   m.Version(),
-			config:    wf.configTree.Get(m.Namespace()),
-		})
+		nss, err := s.metricManager.ExpandWildCards(m.Namespace())
+		if err != nil {
+			// use metric directly from the workflow
+			nss = [][]string{m.Namespace()}
+		}
+
+		for _, ns := range nss {
+			mts = append(mts, &metric{
+				namespace: ns,
+				version:   m.Version(),
+				config:    wf.configTree.Get(m.Namespace()),
```

Comment:
Yes, now it only works for "/intel/mock/*" and for others variations of queries (e.g like /intel/mock/(foo|bar)), but they have to start with "/intel/mock" to get this config. 

I looked on that and I suppose that turning `m.Namespace()` into `ns`  is not the only thing which needs to been done in this case. Look on [scheduler/job.go#L221](https://github.com/intelsdi-x/snap/blob/master/scheduler/job.go#L221) - I do not refactor this file (generally my approach was changing as little as possible), but might this should be changed too - according that, 2 things have to be refactored:
- [in scheduler/job.go func (c *collectorJob) Run()](https://github.com/IzabellaRaulin/snap/blob/dynamic_query_support/scheduler/job.go#L221)
- [in control/control.go func (p *pluginControl) CollectMetrics(..)](https://github.com/IzabellaRaulin/snap/blob/dynamic_query_support/control/control.go#L856)
@jcooklin - what do you think about it?